instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -116,6 +116,11 @@ public static void methodExit(
       callDepth.getAndIncrement();
       try {
         scope.close();
+        Class<? extends HttpURLConnection> connectionClass = connection.getClass();
+
+        String requestMethod = connection.getRequestMethod();
+        GetOutputStreamContext.set(
+            httpUrlState.context, connectionClass, methodName, requestMethod);
```

Comment:
inlining `connectionClass` might be nice to get rid of the generic reference above

Reply:
`I would hazzard to guess that weblogic's HttpURLConnection implementation probably had to conform to the sun behavior `

Maybe , maybe not. We may discuss this point and create a PR outside this issue?  This PR was to fix #4597.